### PR TITLE
Hide Line Numbers switch connected and working

### PR DESCRIPTION
**Pull request by WhitShake**
Added functionality to Line Numbers switch in right navigation panel to hide/show code line numbers in center panel.

### CODE REVIEW

File: src/dotnet/APIView/ClientSPA/angular.json [hunk 1]
```diff
@@ -123,5 +123,8 @@
         }
       }
     }
+  },
```
Comment (chidozieononiwu) [inline src/dotnet/APIView/ClientSPA/angular.json:126]: Can you remove this

File: src/dotnet/APIView/ClientSPA/src/app/_components/code-panel/code-panel.component.html [hunk 2]
```diff
@@ -6,7 +6,7 @@
   <div *uiScroll="let item of codePanelRowSource; let index = index" class="code-line" [attr.data-node-id]="item.nodeIdHashed" [ngClass]="getClassObject(item.rowClasses)">
     <ng-container *ngIf="item.type === CodePanelRowDatatype.CodeLine || item.type === CodePanelRowDatatype.Documentation">
       <div class="line-actions">
-          <span class="line-number first">{{item.lineNumber}}</span>
+          <span *ngIf="!showLineNumbers" class="line-number first">{{item.lineNumber}}</span>
```
Comment (chidozieononiwu) [inline src/dotnet/APIView/ClientSPA/src/app/_components/code-panel/code-panel.component.html:9]: showLineNumbers

File: src/dotnet/APIView/ClientSPA/src/app/_components/code-panel/code-panel.component.ts [hunk 3]
```diff
@@ -21,6 +21,7 @@ export class CodePanelComponent implements OnChanges{
   @Input() navTreeNodIdHashed: string | undefined;
   @Input() reviewId: string | undefined;
   @Input() activeApiRevisionId: string | undefined;
+  @Input() showLineNumbers: boolean = false;
```
Comment (chidozieononiwu) [inline src/dotnet/APIView/ClientSPA/src/app/_components/code-panel/code-panel.component.ts:24]: start out with true

File: src/dotnet/APIView/ClientSPA/src/app/_components/review-page-options/review-page-options.component.ts [hunk 4]
```diff
@@ -53,6 +54,7 @@ export class ReviewPageOptionsComponent implements OnInit, OnChanges{
     this.showCommentsSwitch = this.userProfile?.preferences.showComments ?? true;
     this.showSystemCommentsSwitch = this.userProfile?.preferences.showSystemComments ?? true;
     this.showDocumentationSwitch = this.userProfile?.preferences.showDocumentation ?? false;
+    this.showLineNumbersSwitch = !(this.userProfile?.preferences.showDocumentation ?? false);
```
Comment (chidozieononiwu) [inline src/dotnet/APIView/ClientSPA/src/app/_components/review-page-options/review-page-options.component.ts:57]:
```suggestion
    if (this.userProfile?.preferences.hideLineNumber){
       this.showLineNumbersSwitch = false
    }
```

File: src/dotnet/APIView/ClientSPA/src/app/_components/review-page-options/review-page-options.component.ts [hunk 5]
```diff
@@ -64,6 +66,7 @@ export class ReviewPageOptionsComponent implements OnInit, OnChanges{
       this.showCommentsSwitch = this.userProfile?.preferences.showComments ?? this.showCommentsSwitch;
       this.showSystemCommentsSwitch = this.userProfile?.preferences.showSystemComments ?? this.showSystemCommentsSwitch;
       this.showDocumentationSwitch = this.userProfile?.preferences.showDocumentation ?? this.showDocumentationSwitch;
+      this.showLineNumbersSwitch = !(this.userProfile?.preferences.hideLineNumbers ?? true);
```
Comment (chidozieononiwu) [inline src/dotnet/APIView/ClientSPA/src/app/_components/review-page-options/review-page-options.component.ts:69]: Same as above

File: src/dotnet/APIView/ClientSPA/angular.json (hunk 1 above)
Comment (WhitShake) [reply src/dotnet/APIView/ClientSPA/angular.json:126]: Removing either/both of these produces a red squiggle under "cli" and a yellow one under "analytics"

  },
  "cli": {
    "analytics": false
  }
}

File: src/dotnet/APIView/ClientSPA/src/app/_components/review-page/review-page.component.ts [hunk 6]
```diff
@@ -217,6 +222,14 @@ export class ReviewPageComponent implements OnInit {
     }
   }
 
+  handleShowLineNumbersEmitter(state: boolean) {
+    let userPreferenceModel = this.userProfile?.preferences;
+    userPreferenceModel!.hideLineNumbers = state;
```
Comment (chidozieononiwu) [inline src/dotnet/APIView/ClientSPA/src/app/_components/review-page/review-page.component.ts:227]:
```suggestion
    userPreferenceModel!.hideLineNumbers = !state;
```

File: src/dotnet/APIView/ClientSPA/src/app/_components/review-page/review-page.component.ts [hunk 7]
```diff
@@ -217,6 +222,14 @@ export class ReviewPageComponent implements OnInit {
...
+    this.userProfileService.updateUserPrefernece(userPreferenceModel!).pipe(takeUntil(this.destroy$)).subscribe(() => {
+      this.showLineNumbers = state;
```
Comment (chidozieononiwu) [inline src/dotnet/APIView/ClientSPA/src/app/_components/review-page/review-page.component.ts:229]:
```suggestion
      this.showLineNumbers = userPreferenceModel!.hideLineNumbers;
```

File: src/dotnet/APIView/ClientSPA/src/app/_components/review-page/review-page.component.ts [hunk 8]
```diff
@@ -59,6 +60,10 @@ export class ReviewPageComponent implements OnInit {
     this.userProfileService.getUserProfile().subscribe(
       (userProfile : any) => {
         this.userProfile = userProfile;
+        console.log('userProfile', userProfile)
```
Comment (chidozieononiwu) [inline src/dotnet/APIView/ClientSPA/src/app/_components/review-page/review-page.component.ts:63]: Please remove

**Review by chidozieononiwu — APPROVED** [review]

**Pull request merged**